main/resource-access.bs

```diff
@@ -75,6 +75,291 @@ Issue: Write Linked Data Platform section.
 Draft:
 A Solid data pod MUST conform to the LDP specification [[!LDP]].
 
+## Resource Metadata ## {#rm}
+
+### Background and Need ### {#rm-need}
+<em>This section is non-normative.</em>
+
+Resource Metadata introduces a mechanism for linking to metadata about resources
+in the Solid Ecosystem using HTTP Link headers. The metadata may serve as
+supplementary descriptions or, when supported by a Solid server implementation,
+may influence the behavior of the resources. A given resource might link to zero
+or more such related metadata representations. It is not meant to supplant the
+ability for the directly associated resource to self-describe.
+
+Examples of this mechanism in use include:
+
+<ul>
+  <li>A binary JPEG image with a Link header, pointing to a description that has
+  an RDF representation.</li>
+  <li>A container with a Link header, pointing to an ACL resource that governs
+  access controls for the contained resources.</li>
+  <li>A resource whose shape is constrained by a ShEx schema includes a Link
+  header that points to a metadata resource defining that constraint.</li>
+  <li>A resource with an associated set of configurable parameters includes a
+  Link header that points to a resource where those configurable parameters
+  reside.</li>
+</ul>
+
+### Metadata Discovery ###  {#rm-discovery}
+
+To discover the metadata resources directly associated with a given target
+resource, a client MUST issue a `HEAD` or `GET` request to the target resource
+URL and inspect the `Link` headers in the response, or make an HTTP `GET`
+request on the target resource URL to retrieve an RDF representation, and
+inspect the encoded RDF graph for relations of one or more metadata types. These
+may be carried out in either order, but if the first fails to result in
+discovering the metadata resource or the described resource, the second MUST be
```

Comment:
I've explained MUST, objective, cases with certain kinds... at https://github.com/solid/data-interoperability-panel/pull/32#issuecomment-599977210 . A client might not care about what exactly? If they're engaged in discovery, they certainly do care to follow through and check possible places. Of course they can halt the process at any point (as explained in panel PR comment) but the point is about capability.

"the" metadata would be a reference to reserved metadata types (as outlined re acl, shape, configuration, server managed, describedby).

Client "fails" to discover simply by not finding the desired relation. There is no error code for this kind of discovery, if that's what you're looking for.

Here the minimal expectation between the Link header relations and the RDF body relations would be that they shouldn't conflict eg. don't use different target URIs. A server can only control the headers but probably should stay out of the RDF body that's controlled by a client. Naturally that introduces different complications. See below.

---

Going forward:

I've explained the background on my proposal for documentation and then retracted the proposal in https://github.com/solid/data-interoperability-panel/issues/37#issuecomment-603110090 in response to @ericprud 's points. The limitations or perhaps the expectations it sets on server needs more attention in my opinion. If only the server can set the Link headers for metadata resource discovery, we need to be clear on the conditions in which each relation (per metadata type) is revealed by a server.

Reply:
Agree with @RubenVerborgh's [suggestion](https://github.com/solid/specification/pull/156/files/4c4dde9fe6352dd5802fd4b4b4d0fc98c7de7657#r396715703) that we make the exposure / discovery something that is specific to metadata types. I think we adjust this introductory text to enumerate the different ways of exposure and discovery (link headers, rdf graph), and then get specific for each type. Following that, we relax the MUST so that a client isn't required to look for every possible type of metadata in each path, each time.

> If only the server can set the Link headers for metadata resource discovery, we need to be clear on the conditions in which each relation (per metadata type) is revealed by a server.

@csarven The description of the different types already is doing a fair bit of this, but please take a look at comment anywhere you feel is lacking.

Reply:
Have cleared out the requirement for inspection of the RDF body for metadata in general discovery. Since the scope has been narrowed from overall resource metadata to auxiliary resources, we can leave it up to the a given auxiliary resources types to define whether it is necessary to check the RDF body as part of processing.